### PR TITLE
Check for header auth token on persistent file download

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose

With the addition of the new shiny persistent file download route (#8107), users couldn't provide an oauth token as they used to in order to authenticate for download requests. Let's fix that!

## Changes

- add check in `_get_current_user` method to look for token. This is eventually called from the `@collect_auth` decorator
- test for getting with token in header

## QA Notes

I do believe that @TimErrington was the one who reported this bug because of a script he runs regularly, so perhaps this would be best tested by him!

## Side Effects

Any route using the `get_user` method from `core.py` can now pass a token in the header, which is nice!

## Ticket

https://openscience.atlassian.net/browse/PLAT-741

### CODE REVIEW

File: framework/auth/core.py [hunk 1]
```diff
@@ -177,7 +177,15 @@ def private_link(self):
 
     @classmethod
     def from_kwargs(cls, request_args, kwargs):
-        user = request_args.get('user') or kwargs.get('user') or _get_current_user()
+        from osf.models import ApiOAuth2PersonalToken
```
Comment (sloria) [inline framework/auth/core.py]: Drive-by: Do both of these need to be imported here?

File: framework/auth/core.py [hunk 2]
```diff
@@ -177,7 +177,15 @@ def private_link(self):
...
+        from framework.auth.cas import parse_auth_header
+
+        header_token = request.headers.get('Authorization', None)
```
Comment (sloria) [inline framework/auth/core.py]: Drive-by: This will fail if `from_kwargs` is used outside of a request context. This *probably* isn't a blocker: `from_kwargs` *should* only be called within a request, but I'm not sure if that's true without doing a code dive. 

Ideally, we wouldn't be reading from the request at all here (read from the arguments instead), but I can see why you did it that way.

Comment (sloria) [reply framework/auth/core.py]: Ah, just had a thought: what if you modified `_get_current_user` to check the bearer token instead of doing it here? That feels like a better place for this logic, since that function is already reading from the request.

**Comment by TimErrington** [comment]
Yes, happy to test this when ready. Just let me know

File: framework/auth/core.py (hunk 1 above)
Comment (erinspace) [reply framework/auth/core.py]: Yea sadly, importing at top of `core.py` gives a circular import error:
```
File "/Users/erin/code/osf_stuff/osf.io/addons/osfstorage/apps.py", line 1, in <module>
    from addons.base.apps import BaseAddonAppConfig
  File "/Users/erin/code/osf_stuff/osf.io/addons/base/apps.py", line 11, in <module>
    from website.util import rubeus
  File "/Users/erin/code/osf_stuff/osf.io/website/util/rubeus.py", line 11, in <module>
    from framework.auth.decorators import Auth
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/__init__.py", line 6, in <module>
    from framework.auth.core import Auth
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/core.py", line 13, in <module>
    from framework.auth.cas import parse_auth_header
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/cas.py", line 11, in <module>
    from framework.auth import authenticate, external_first_login_authenticate
ImportError: cannot import name authenticate
```

**Comment by sloria** [comment]
Good to merge when Travis shows green.
